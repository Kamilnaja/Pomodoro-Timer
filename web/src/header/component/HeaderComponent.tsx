import React from 'react';
import { Container, Nav, Navbar } from 'react-bootstrap';
import { BrowserRouter as Router, Route, Switch } from 'react-router-dom';
import { ModalType } from 'shared/modal/modalEnum';
import { About } from '../../about/About';
import { ConnectedPomodoroCounterScreen } from '../../pomodoroCounter/pomodoroCounterScreen/PomodoroCounterScreen';
import { SettingsContainer } from '../../settings/container/SettingsContainer';
import { StatsScreenContainer } from '../../stats/statsScreen/StatsScreenContainer';
import { HeaderContainerProps } from '../container/HeaderContainerProps';
import { GuardedRoute } from '../guardedRoute/GuardedRoute';
import './headerComponent.scss';

export const HeaderComponent = (props: HeaderContainerProps) => (
  <Router>
    <Container>
      <Navbar>
        <Navbar.Brand href="/">Tomatodone</Navbar.Brand>
        <Navbar.Toggle aria-controls="basic-navbar-nav" />
        <Navbar.Collapse id="basic-navbar-nav">
          <Nav className="mr-auto">
            {props.isLoggedIn ? (
              <>
                <Nav.Link href="/stats">Stats</Nav.Link>
                <Nav.Link href="/settings">Settings</Nav.Link>
                <Nav.Link className="button--logout" onClick={() => props.handleOpenModal(ModalType.LOGOUT)}>
                  Logout
                </Nav.Link>
              </>
            ) : (
              <>
                <Nav.Link className="button--login" onClick={() => props.handleOpenModal(ModalType.LOGIN)}>
                  Login
                </Nav.Link>
                <Nav.Link className="button--register" onClick={() => props.handleOpenModal(ModalType.REGISTER)}>
                  Register
                </Nav.Link>
              </>
            )}
            <Nav.Link href="/about">About</Nav.Link>
          </Nav>
        </Navbar.Collapse>
      </Navbar>
      <Switch>
        <GuardedRoute path="/settings" component={SettingsContainer} auth={true}></GuardedRoute>
        <GuardedRoute path="/stats" component={StatsScreenContainer} auth={props.isLoggedIn}></GuardedRoute>
        <Route path="/about">
          <About />
        </Route>
        <Route path="/">
          <ConnectedPomodoroCounterScreen />
        </Route>
      </Switch>
    </Container>
  </Router>
);
