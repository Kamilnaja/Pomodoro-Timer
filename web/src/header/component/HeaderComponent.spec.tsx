import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import { HeaderComponent } from './HeaderComponent';

describe('HeaderComponent', () => {
  const component = shallow(<HeaderComponent handleOpenModal={() => {}} isLoggedIn={true} />);

  it('Should match snapshot', () => {
    expect(shallowToJson(component)).toMatchSnapshot();
  });
});
