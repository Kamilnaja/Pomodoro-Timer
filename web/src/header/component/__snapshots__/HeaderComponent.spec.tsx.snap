// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HeaderComponent Should match snapshot 1`] = `
<BrowserRouter>
  <Container
    fluid={false}
  >
    <Navbar
      collapseOnSelect={false}
      expand={true}
      variant="light"
    >
      <NavbarBrand
        href="/"
      >
        Tomatodone
      </NavbarBrand>
      <NavbarToggle
        aria-controls="basic-navbar-nav"
        label="Toggle navigation"
      />
      <NavbarCollapse
        id="basic-navbar-nav"
      >
        <Nav
          className="mr-auto"
          fill={false}
          justify={false}
        >
          <NavLink
            as={
              Object {
                "$$typeof": Symbol(react.forward_ref),
                "render": [Function],
              }
            }
            disabled={false}
            href="/stats"
          >
            Stats
          </NavLink>
          <NavLink
            as={
              Object {
                "$$typeof": Symbol(react.forward_ref),
                "render": [Function],
              }
            }
            disabled={false}
            href="/settings"
          >
            Settings
          </NavLink>
          <NavLink
            as={
              Object {
                "$$typeof": Symbol(react.forward_ref),
                "render": [Function],
              }
            }
            disabled={false}
            onClick={[Function]}
          >
            Logout
          </NavLink>
        </Nav>
      </NavbarCollapse>
    </Navbar>
    <Switch>
      <Route
        path="/taskAndNotes"
      >
        <Connect(TaskAndNotesContainer) />
      </Route>
      <Route
        path="/settings"
      >
        <SettingsContainer />
      </Route>
      <Route
        path="/stats"
      >
        <Connect(StatsContainer) />
      </Route>
      <Route
        path="/"
      >
        <Connect(PomodoroCounterScreen) />
      </Route>
    </Switch>
  </Container>
</BrowserRouter>
`;
